tgui/packages/tgui/interfaces/ZoldorfPlayerShop/index.tsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Glowbold (https://github.com/pgmzeta)
+ * @license MIT
+ */
+
+import { useBackend } from "../../backend";
+import { Box, Button, Flex, Stack, Table } from "../../components";
+import { Window } from "../../layouts";
+import { ZoldorfPlayerShopData, ZoldorfProductListProps } from "./type";
+
+export const ZoldorfPlayerShop = (_, context) => {
+  const { act, data } = useBackend<ZoldorfPlayerShopData>(context);
+  const { soul_products, credit_products, credits } = data;
+  return (
+    <Window
+      width="500"
+      height="600"
+      fontFamily="Consolas"
+      font-size="10pt">
+      <Window.Content>
+        <Stack vertical fill minHeight="1%" maxHeight="100%">
+          <Stack.Item grow minHeight="1%" maxHeight="100%">
+            <Stack vertical>
+              {
+                soul_products.map(product => {
+                  return (
+                    <ZoldorfProductList {...product} key={product.name}>
+                      <Button
+                        color="red"
+                        onClick={() => act('soul_purchase', { "item": product.name })}
```

Comment:
Would be nice if buttons were uniform in size
![image](https://github.com/goonstation/goonstation/assets/757551/e127edad-d39f-4e66-a68c-c52548d4578d)